README.md

```diff
@@ -32,3 +28,152 @@
   1. Socket.io and Redis
   2. Popular modules
   3. Links
+
+## Introduction to NodeJS ##
+### CommonJS Modules ###
+Code in node apps is organized in modules. There are a couple of standards on how to write a module in JavaScript. By convention, NodeJS has decided to use the CommonJS Module Specification.
+
+* [Module 1.0 - CommonJS Specification](http://www.commonjs.org/specs/modules/1.0/ "Module 1.0 - CommonJS Specification")
+* [A complete explanation of the 3 most used module definitions](http://addyosmani.com/writing-modular-js/ "Modules in JavaScript")
+### Intro to npm ###
+npm is a package manager for JavaScript, and is the default for Node.js. npm makes it easy for JavaScript developers to share and reuse code, and it makes it easy to update the code that you're sharing.
+
+* [What is npm? Short explanation video](https://www.youtube.com/watch?v=pa4dc480Apo)
+* [Installing node and updating npm](https://docs.npmjs.com/getting-started/installing-node)
+* [Create and publish modules to npm](https://quickleft.com/blog/creating-and-publishing-a-node-js-module/)
+
+### Globals, Process and Buffer###
+As we have learnt, node's module system discourages the use of globals; however node provides a few important globals for use to utilize. Here we list the most important ones.
+
+- [Buffer](http://nodejs.org/api/buffer.html): To handle binary data, node provides us with the global Buffer object. Buffer instances represent memory allocated independently of V8's heap. [Here](http://www.w3resource.com/node.js/nodejs-buffer.php) is an example of how to use buffers.	 
```

Comment:
Change "node provides us with the global Buffer object" to "NodeJS provides the global Buffer object"